linera-service/src/benchmark.rs

```diff
@@ -0,0 +1,266 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use clap::Parser as _;
+use fungible::{Account, AccountOwner, FungibleTokenAbi, InitialState, Parameters};
+use futures::future::join_all;
+use linera_base::{
+    async_graphql::InputType,
+    data_types::Amount,
+    identifiers::{ApplicationId, ChainId, Owner},
+};
+use linera_execution::system::SystemChannel;
+use linera_service::cli_wrappers::{ApplicationWrapper, ClientWrapper, Network};
+use port_selector::random_free_tcp_port;
+use rand::{Rng as _, SeedableRng};
+use serde_json::Value;
+use std::{collections::BTreeMap, path::Path, sync::Arc, time::Duration};
+use tempfile::tempdir;
+use tokio::time::Instant;
+
+#[derive(clap::Parser)]
+#[command(
+    name = "linera-benchmark",
+    version = clap::crate_version!(),
+    about = "Run benchmarks against a Linera network",
+)]
+enum Args {
+    Fungible {
+        /// The number of users in the test.
+        #[arg(long = "users", default_value = "4")]
+        users: usize,
+
+        /// The number of transactions being made per user.
+        #[arg(long = "transactions", default_value = "4")]
+        transactions: usize,
+
+        /// The faucet (which implicitly defines the network)
+        #[arg(long = "faucet", default_value = "http://faucet.devnet.linera.net")]
+        faucet: String,
+
+        /// The seed for the PRNG determining the pattern of transactions.
+        #[arg(long = "seed", default_value = "0")]
+        seed: u64,
+
+        #[arg(long = "uniform")]
+        /// If set, each chain receives the exact same amount of transfers.
+        uniform: bool,
+    },
+}
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let args = Args::parse();
+    match args {
+        Args::Fungible {
+            users,
+            transactions,
+            faucet,
+            seed,
+            uniform,
+        } => benchmark_with_fungible(users, transactions, faucet, seed, uniform).await,
+    }
+}
+
+async fn benchmark_with_fungible(
+    n_users: usize,
```

Comment:
I think we're writing `num_users` in the rest of the codebase.

Reply:
`num_wallets` even may be more precise.